addons/mute-project/userscript.js

```diff
@@ -1,37 +1,65 @@
 export default async function ({ addon, global, console }) {
   const vm = addon.tab.traps.vm;
-  let muted = false;
+  const muteIcon = "/static/assets/e21225ab4b675bc61eed30cfb510c288.svg";
+  const quietIcon = "/static/assets/3547fa1f2678a483a19f46852f36b426.svg";
+  const loudIcon = "/static/assets/b2c44c738c9cbc1a99cd6edfd0c2b85b.svg";
+  let showSlider = addon.settings.get("show-slider");
   let icon = document.createElement("img");
-  icon.src = "/static/assets/e21225ab4b675bc61eed30cfb510c288.svg";
   icon.loading = "lazy";
-  icon.style.display = "none";
+  let slider = document.createElement("input");
+  slider.type = "range";
+  slider.min = 0;
+  slider.max = 1;
+  slider.step = 0.02;
   const toggleMute = (e) => {
     if (!addon.self.disabled && (e.ctrlKey || e.metaKey)) {
       e.cancelBubble = true;
       e.preventDefault();
-      muted = !muted;
-      if (muted) {
-        vm.runtime.audioEngine.inputNode.gain.value = 0;
-        icon.style.display = "block";
-      } else {
-        vm.runtime.audioEngine.inputNode.gain.value = 1;
-        icon.style.display = "none";
-      }
+      setVol(slider.value == 0 ? 1 : 0);
     }
   };
+
+  function setVol(v) {
+    vm.runtime.audioEngine.inputNode.gain.value = v;
+    slider.value = v;
+    slider.style.display = showSlider ? "inline-block" : "none";
+    if (v == 0) {
+      icon.src = muteIcon;
+      icon.style.display = "inline-block";
+      return;
+    }
+    icon.src = v < 0.5 ? quietIcon : loudIcon;
+    icon.style.display = showSlider ? "inline-block" : "none";
+  }
+
   addon.self.addEventListener("disabled", () => {
-    muted = false;
     vm.runtime.audioEngine.inputNode.gain.value = 1;
-    icon.style.display = "none";
+  });
+
+  addon.self.addEventListener("reenabled", () => {
+    setVol(1);
   });
 
   while (true) {
     let button = await addon.tab.waitForElement("[class^='green-flag_green-flag']", {
       markAsSeen: true,
       reduxEvents: ["scratch-gui/mode/SET_PLAYER", "fontsLoaded/SET_FONTS_LOADED", "scratch-gui/locales/SELECT_LOCALE"],
     });
-    addon.tab.appendToSharedSpace({ space: "afterStopButton", element: icon, order: 0 });
+    const container = document.createElement("div");
+    container.className = "sa-volume";
+    addon.tab.displayNoneWhileDisabled(container, { display: "inline-block" });
+    addon.tab.appendToSharedSpace({ space: "afterStopButton", element: container, order: 0 });
+    container.appendChild(icon);
+    container.appendChild(slider);
     button.addEventListener("click", toggleMute);
     button.addEventListener("contextmenu", toggleMute);
+    setVol(1);
+    slider.addEventListener("change", function (e) {
+      setVol(this.value);
+    });
```

Comment:
Is it possible to update the volume live while the user moves the slider, instead of waiting until mouseup (releasing click)?